scripts/validate.py

```
@@ -474,37 +455,52 @@ def toml_string(path: str, table: str, key: str) -> str:
-        if version != expected:
-            errors.append(f"{path}: version {version!r} != {expected!r}")
+    actual: dict[str, str] = {}
+    for manifest in JSON_MANIFESTS:
+        actual.update(json_versions(manifest))
+
+    # The CHANGELOG's newest heading is a version string too, and was the one
+    # place nothing checked: it could say 0.2.0 while every manifest said 0.3.0.
+    changelog = (root / "CHANGELOG.md").read_text(encoding="utf-8")
+    heading = re.search(rf"^##\s+(?P<version>{VERSION_PATTERN})", changelog, re.MULTILINE)
+    if heading is None:
+        errors = [f"CHANGELOG.md: no '## <version>' heading found"]
```

Ruff (0.16.2)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)